src/Platform/Microsoft.Testing.Platform/AI/IChatClientManager.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Platform.Services;
+
+namespace Microsoft.Testing.Platform.AI;
+
+internal interface IChatClientManager
+{
+    void SetChatClientProviderFactory(Func<IServiceProvider, object> chatClientProviderFactory);
```

Comment:
To match other API it's probably better as `AddChatClientProvider`, ideally having a marker interface instead of `object`.